File: config/teacherGen.py
import csv, time
from random import choice

numTeachers = 1000000

subjects = [
    ["Mathematics",0,0,0],
    ["English Language",
    "English Literature",0,0],
    ["Biology",0,0,0],
    ["Chemistry",0,0,0],
    ["Physics",0,0,0],
    ["Religious Studies",0,0,0],
    ["Physical Education",0,0,0],
    ["Computer Science",0,0,0],
    ["History",0,0,0],
    ["Geography",0,0,0],
    ["Modern Language",
    "French",
    "German",
    "Spanish"],
    ["Drama & Theatre Studies",0,0,0],
    ["Music",0,0,0],
    ["Electronics",0,0,0],
]

start = time.perf_counter()

with open('teachers.csv','w') as csvfile:
    writer = csv.writer(csvfile)
    for teacher in range(numTeachers):
        teacherID = [f'teacher{teacher}']
        teacherSubjects = choice(subjects)
        writer.writerow(teacherID + teacherSubjects)

elapsed = time.perf_counter() - start
print(f"Took {elapsed:.3f}s")
        

File: config/teacherGen.rs
use csv::Writer;
use rand::seq::SliceRandom;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let num_teachers = 10;

    let subjects = vec![
        vec!["Mathematics", "", "", ""],
        vec!["English Language", "English Literature", "", ""],
        vec!["Biology", "", "", ""],
        vec!["Chemistry", "", "", ""],
        vec!["Physics", "", "", ""],
        vec!["Religious Studies", "", "", ""],
        vec!["Physical Education", "", "", ""],
        vec!["Computer Science", "", "", ""],
        vec!["History", "", "", ""],
        vec!["Geography", "", "", ""],
        vec!["Modern Language", "French", "German", "Spanish"],
        vec!["Drama & Theatre Studies", "", "", ""],
        vec!["Music", "", "", ""],
        vec!["Electronics", "", "", ""],
    ];

    let mut rng = rand::thread_rng();
    let mut writer = Writer::from_path("teachers.csv")?;

    for teacher in 0..num_teachers {
        let mut row = vec![format!("teacher{}", teacher)];

        let chosen = subjects.choose(&mut rng).unwrap();
        row.extend(chosen.iter().map(|s| s.to_string()));

        writer.write_record(&row)?;
    }

    writer.flush()?;
    Ok(())
}